### PR TITLE
.github: secure github actions

### PR DESCRIPTION
**Pull request by tuminoid**
Set top-level permissions to read.

Pin actions by a sha.

### CODE REVIEW

File: .github/workflows/kubesec.yml [hunk 1]
```diff
@@ -13,7 +16,7 @@ jobs:
       contents: read
```
Comment (kashifest) [inline .github/workflows/kubesec.yml:16]: Curious to know why do we need top level permission set if we have set it here in more granular steps.

Comment (kashifest) [reply .github/workflows/kubesec.yml:16]: So why not `permissions: {} `at the top level and let the jobs decide what it needs?

Comment (tuminoid) [reply .github/workflows/kubesec.yml:16]: Avoids repetition. But it could be done like that too.

Comment (kashifest) [reply .github/workflows/kubesec.yml:16]: I am fine with the current change as well. Was curious to know the impact of different configurations.
/approve

**Comment by tuminoid** [comment]
> You can actually change the default behavior for all workflows in settings only. This makes sure none of the workflows would have write access by default. This can be done in org level as well as individual repo level.

They can be set, but then they're decoupled from actual actions. The further away you set them (repo or org level), the more likely it is that there is a gap. Setting them in the workflow files keeps them up to date with the jobs.

Thanks for the discussion! It is good that changes are not rubber-stamped, but well scrutinized.

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Pull request merged**